### ISSUE TITLE
distutils/setuptools integration

### ISSUE THREAD

**Issue by spooning** [robotframework/robotframework#1685]
> _Originally submitted to [Google Code](http://code.google.com/p/robotframework/issues/detail?id=1685) by michael.walle on 21 Mar 2014_

It would be cool to have a distutils command to build library documentation. Eg. like the "build_sphinx" subcommand, something like build_rf_libdoc.

This will ease the packaging of third-party libraries.

If i understand it correctly, this command - if implemented - should then be automatically available for all distutils/setuptools-style setup.py files.

**Comment by pekkaklarck** [comment]
@mwalle, are you still interested in this?

**Comment by mwalle** [comment]
sure!

**Comment by pekkaklarck** [comment]
@mwalle, was your idea to enhance setuptools integration in general or just add this entry point to create library documentation? In the former case, could you clarify what other ideas you have? Could you also explain use cases for this kind of library documentation generation? What benefits would it have compared to having a separate step in the build process to generate documentation using Libdoc normally?

**Comment by pekkaklarck** [comment]
Ping @mwalle! Still interested about this issue? Notice that we nowadays use setuptools when available and also plan to start using entry points to implement `robot.exe` in RF 3.1 (#2415).

**Comment by mwalle** [comment]
Yes still interested.. I've just tried this patch on the latest master branch. I will provide a pull request.

> was your idea to enhance setuptools integration in general or just add this entry point to create library documentation?

I don't understand what you mean. The idea is to provide a setuptools command extension and if a user has robotframework installed, he can generate a documentation by just use ``setup.py build_rf_docs`` (without the need to modify the setup.py file). I'll give you an example:

```shell
$ git clone https://github.com/robotframework/SSHLibrary.git
$ cd SSHLibrary
$ ./setup.py build_rf_docs
running build_rf_docs
/home/mwalle/repos/SSHLibrary/docs/SSHLibrary.html
```

Please note, that I haven't modified any files within the library repository.

```shell
$ ./setup.py --help-commands
Standard commands:
  build             build everything needed to install
  build_py          "build" pure Python modules (copy to build directory)
  build_ext         build C/C++ extensions (compile/link to build directory)
  build_clib        build C/C++ libraries used by Python extensions
  build_scripts     "build" scripts (copy and fixup #! line)
  clean             clean up temporary files from 'build' command
  install           install everything from build directory
  install_lib       install all Python modules (extensions and pure Python)
  install_headers   install C/C++ header files
  install_scripts   install scripts (Python or otherwise)
  install_data      install data files
  sdist             create a source distribution (tarball, zip file, etc.)
  register          register the distribution with the Python package index
  bdist             create a built (binary) distribution
  bdist_dumb        create a "dumb" built distribution
  bdist_rpm         create an RPM distribution
  bdist_wininst     create an executable installer for MS Windows
  upload            upload binary package to PyPI
  check             perform some checks on the package

Extra commands:
  [..]
  build_rf_docs     Build Robotframework library documentation
  [..]
```

Maybe one can come up with a better name than ``build_rf_docs``, that is acutally borrowed from sphynx where the command is ``build_sphinx``.


> What benefits would it have compared to having a separate step in the build process to generate documentation using Libdoc normally?

What is needed for the seperate step? Providing this extension, it is very easy for the user/developer to build library documention. And if i remember correctly, you can just add the ``build_rf_docs` to your build steps (I don't know the exact file/place at the moment), so one doesn't have to open code the library generation anymore.
